src/libaktualizr/utilities/types_test.cc

```diff
@@ -31,6 +31,20 @@ TEST(TimeStamp, Now) {
   EXPECT_LT(t_now, t_future);
 }
 
+TEST(Types, ResultCode) {
+  data::ResultCode ok_res{data::ResultCode::Numeric::kOk};
+  EXPECT_EQ(ok_res.num_code, data::ResultCode::Numeric::kOk);
+  EXPECT_EQ(ok_res.toString(), "OK");
+  std::string repr = ok_res.toRepr();
+  EXPECT_EQ(repr, "OK:0");
+  EXPECT_EQ(data::ResultCode::fromRepr(repr), ok_res);
+
+  // !
+  EXPECT_NE(ok_res, data::ResultCode(data::ResultCode::Numeric::kOk, "OK2"));
+  EXPECT_NE(ok_res, data::ResultCode(data::ResultCode::Numeric::kGeneralError, "OK"));
+  EXPECT_EQ(data::ResultCode::fromRepr("OK"), data::ResultCode(data::ResultCode::Numeric::kUnknown, "OK"));
```

Comment:
What's with the comment? Is there something unexpected here?

Reply:
The comparison operator could appear surprising. For example, if there is no `:k` part, the numeric code is unknown. These are three instances of gotcha comparison, but the comment could be more verbose I admit.

Reply:
It looks fine to me, I was just alarmed by the exclamation point. A better comment may be helpful, but it's fine as it is.